Add IPFS Update instructions to the Install section.

alexmmueller opened the pull request:
It is not completely ready by now, but I thought I get your comments as its almost finished.

* Uninstalll commands for windows are missing, but section is prepared. Maybe you can add them?
* I tried to use the big OS icon links in the `ipfs-update` section, but due to OS same section names it is not possible to link to the correct sections. I skipped it for now. Please let me know if you prefer something els
* I used `curl` in instructions for macOS as `wget` is not preinstalled on macOs (at least not on my boxes), but `curl` is
* I added `sudo` to the commands where root privileges are required

Feedback is welcome :)

Admin edit: Closes #300

johnnymatthews commented on docs/install/command-line.md:
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+Note: Your $GOPATH/bin should be within $PATH for the result ipfs-update binary
+to be found.
+
+### Use `ipfs-update` to install & update Go-IPFS
```
```suggestion
### Install using IPFS Update
```

johnnymatthews commented on docs/install/command-line.md:
````diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+
+### Use `ipfs-update` to install & update Go-IPFS
+
+```bash
+$ ipfs-update install <version>
+```
````
````suggestion
Run `ipfs-update install` followed by the version of Go-IPFS you want to install:

```bash
ipfs-update install 0.5.0
```

To install the latest release of Go-IPFS use the `latest` tag:

```bash
ipfs-update install latest
```

````

johnnymatthews commented on docs/install/command-line.md:
````diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+$ ipfs-update install <version>
+```
+
+Downloads, tests, and installs the specified version (or "latest" for latest version) of Go-IPFS. The existing version is stashed in case a revert is needed.
````
```suggestion
`ipfs-update install` downloads, tests, and installs the specified version of Go-IPFS. If a version of IPFS is already installed, that version is _stashed_ and can be reverted to later.
```

johnnymatthews commented on docs/install/command-line.md:
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+
+Downloads, tests, and installs the specified version (or "latest" for latest version) of Go-IPFS. The existing version is stashed in case a revert is needed.
+
+Note: In order to update Go-IPFS just install the new version. Check See the [`ipfs/ipfs-update` GitHub repository for details →](https://github.com/ipfs/ipfs-update)
```
```suggestion
```

johnnymatthews commented on docs/install/command-line.md:
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+
+Note: In order to update Go-IPFS just install the new version. Check See the [`ipfs/ipfs-update` GitHub repository for details →](https://github.com/ipfs/ipfs-update)
+
+### Use `ipfs-update` to downgrade Go-IPFS
```
```suggestion
### Downgrade using IPFS Update
```

johnnymatthews commented on docs/install/command-line.md:
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+Note: In order to update Go-IPFS just install the new version. Check See the [`ipfs/ipfs-update` GitHub repository for details →](https://github.com/ipfs/ipfs-update)
+
+### Use `ipfs-update` to downgrade Go-IPFS
+
```
```suggestion
Use the `revert` function to roll-back to a previous version of Go-IPFS.

```

johnnymatthews commented on docs/install/command-line.md:
````diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+### Use `ipfs-update` to downgrade Go-IPFS
+
+```bash
+$ ipfs-update revert
````
```suggestion
ipfs-update revert
```

johnnymatthews commented on docs/install/command-line.md:
````diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+$ ipfs-update revert
+```
+
+Reverts to the previously installed version of Go-IPFS. This is useful if the newly installed version has issues and you would like to switch back to your older stable installation.
````
```suggestion
`ipfs-update revert` reverts to the previously installed version of Go-IPFS. This is useful if the newly installed version has issues and you would like to switch back to your older stable installation.
```

johnnymatthews commented on docs/install/command-line.md:
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
+
+### Uninstall `ipfs-update`
+
+Uninstalling `ipfs-update` involves removing the binary and adjusting the `PATH` variable on Windows systems.
```
```suggestion
To uninstall IPFS Update, delete the binary and `ipfs-update` from your `PATH` variable.
```

johnnymatthews reviewed (changes requested):
This is a great start! However, I think it might be better if we moved everything in here into it's own page. Somewhere like `install/ipfs-update`. This `install/command-line` page is already getting pretty long.

Carry on with your work here though, and I'll move things around once you've finished :)

Also, I've converted this PR to a draft since you're still working on it.

alexmmueller commented:
I accepted all changes you proposed @johnnymatthews. I am a little unsre of the value of having powershell command for updating `PATH`as this is not trival it seams and requires a small program (get variable, iterate over it, etc). Finding a file is a little easier: Can you check if the powershell command:

gci -recurse -filter "ipfs-update.exe" -File -ErrorAction SilentlyContinue

Will work? If not, I will setup a VM to try this out.

johnnymatthews commented:
Sounds good. If you've got time to explore the Powershell section through a VM then that'd be great! No worries if you can't though. I've got a Windows machine, but it's out of action right now. So it'll be a couple of days until I can get into it.

> Could I have create the PR as a draft as well?

Yeah! I didn't realize this for a while, but there's a drop down menu when you click **Create pull request** that lets you create the PR as a draft. [It's kinda hidden](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

johnnymatthews commented:
Looks like my Windows machine is gonna be out of action for a while, so I'll boot up a VM and get the Powershell stuff tested.

alexmmueller commented:
> Were you able to boot up a Windows VM @alexmmueller?

After some painful hours, yes :) I guess I will be able to get everything tested within the next weekend. At the moment I am not sure whether to list the commands for removing a variable from PATH in powershell or go for the instructions on how to remove the variable via the settings widget in windows. Removing the directory from PATH via powershell is not a one liner. Do you have any preference @johnnymatthews ?

johnnymatthews commented:
@alexmmueller you can ignore instructions for removing the PATH variable for Windows. There's a bunch of ways to do it, and they all have several steps involved. Instead we should link to the [Microsoft documentation on how to update/edit/delete your PATH](https://docs.microsoft.com/en-us/cpp/build/setting-the-path-and-environment-variables-for-command-line-builds?view=msvc-160).

alexmmueller commented:
I reviewed the windows instructions today and found some issues:
* the PowerShell profile file does not necessarily exist --> I added a check and profile creation instructions
* PowerShell does by default block profile files --> I added a note and a link to the respective documentation on PowerShell ExecutionPolicies

If you want anything changed, let me know.

If you like the changes and there are no change requests, I think we should also adopt the windows PATH part for GO-IPFS accordingly. Do you agree?

johnnymatthews commented on docs/install/command-line.md:
````diff
@@ -321,20 +339,18 @@ To uninstall IPFS Update, delete the binary and `ipfs-update` from your `PATH` v
-   ```powershell
-
-   ```
+3. Update the `PATH` variable as described in the [Microsoft documentation on how to update/edit/delete your PATH](https://docs.microsoft.com/en-us/cpp/build/setting-the-path-and-environment-variables-for-command-line-builds?view=msvc-160)
````
```suggestion
3. Update your systems `PATH` variable. This process differs between Windows installations, so please check the [Microsoft documentation for details](https://docs.microsoft.com/en-us/cpp/build/setting-the-path-and-environment-variables-for-command-line-builds?view=msvc-160).
```

johnnymatthews reviewed (changes requested):
Everything looks really good here! Just one minor change.

alexmmueller commented:
I did the requested change. Let me know if anything else has to be done here.

johnnymatthews commented:
Everything looks good here @alexmmueller, thanks for your efforts! I'm gonna move things around a little bit, and then get things merged in.
